quadriga/src/main/java/edu/asu/diging/quadriga/web/ExploreCollectionController.java

```diff
@@ -0,0 +1,163 @@
+    private ConceptService conceptService;
+  
+    @RequestMapping(value = "/auth/collections/{collectionId}/explore")
+    public String exploreTriples(@PathVariable String collectionId, Model model) {
+        Collection collection;
+        try {
+            collection = collectionManager.findCollection(collectionId);
+        } catch (InvalidObjectIdException e) {
+            logger.error("No Collection found for id {}", collectionId);
+            return "error404Page";
+        }
+        model.addAttribute("collectionName", collection.getName());
+        model.addAttribute("collection", collectionId);
+        return "auth/exploreCollection";
+    }
+
+    @GetMapping(value = "/auth/collections/{collectionId}/sub-graph")
+    public ResponseEntity<GraphElements> getGraphForUri(@PathVariable String collectionId,
+            @RequestParam(value = "uri", required = true) String uri,
+            @RequestParam(value = "ignoreList", required = false, defaultValue = "{}") List<String> ignoreList)
+            throws InvalidObjectIdException,CollectionNotFoundException{
+        GraphElements graphElements = new GraphElements();
+        try {
+            ConceptCache conceptCache = conceptCacheService.getConceptByUri(uri);
+            MappedTripleGroup mappedTripleGroup = mappedTripleGroupService.findByCollectionIdAndMappingType(collectionId, MappedTripleType.DEFAULT_MAPPING);
+            List<DefaultMapping> triples= mappedTripleService.getTriplesByUri(mappedTripleGroup.get_id().toString(),
+                    mapConceptUriToDatabaseUri(mappedTripleGroup.get_id().toString(),conceptCache.getEqualTo()), ignoreList);
+            
+            graphElements = GraphUtil.mapToGraph(triples);
+        }
+        catch(InvalidObjectIdException e)
+        {
+            return new ResponseEntity<>(null,HttpStatus.BAD_REQUEST);
+        }
+        catch(CollectionNotFoundException e)
+        {
+            return new ResponseEntity<>(null,HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(graphElements, HttpStatus.OK);
+    }
+
+    private String mapConceptUriToDatabaseUri(String mappedTripleGroupId,List<String> equalTo)
+    {
+        List<Concept> concept = conceptService.findByMappedTripleGroupId(mappedTripleGroupId);
+        List<String> conceptUris = new ArrayList<>();
+        try {
+            for(Concept i:concept){
+                conceptUris.add(i.getUri());
+            }
+            for(String i:equalTo){
+                List<String> listOfUris = processUri(i);
+                listOfUris.retainAll(conceptUris);
+                if(!listOfUris.isEmpty()) {
+                    return listOfUris.get(0);
+                }
+            }
+        }
+        catch(NullPointerException e)
+        {
+            logger.error("Couldn't find concept", e);
+        }
+        return null;        
+    }
+    
+    // Normalize the URI prefix and suffix
+    private List<String> processUri(String uri) {
```

Comment:
it's probably a lot easier to just find everything after the protocol up to the last slash (if there is one). That should be a fairly simple regex expression (something like `https?(://.*?)/?$`) and then just add four uris with the protocols/trailing slashes.